src/wallet/wallet.cpp

```diff
@@ -2169,14 +2169,18 @@ bool CWallet::GetNewChangeDestination(const OutputType type, CTxDestination& des
     return true;
 }
 
-int64_t CWallet::GetOldestKeyPoolTime() const
+std::optional<int64_t> CWallet::GetOldestKeyPoolTime() const
 {
     LOCK(cs_wallet);
-    int64_t oldestKey = std::numeric_limits<int64_t>::max();
+    if (m_spk_managers.empty()) {
+        return std::nullopt;
+    }
+
+    std::optional<int64_t> oldest_key{std::numeric_limits<int64_t>::max()};
     for (const auto& spk_man_pair : m_spk_managers) {
-        oldestKey = std::min(oldestKey, spk_man_pair.second->GetOldestKeyPoolTime());
+        oldest_key = std::min(oldest_key, spk_man_pair.second->GetOldestKeyPoolTime());
```

Comment:
Couldn't you get a nullopt in here? (If not, why change this block at all?)

Reply:
> Couldn't you get a nullopt in here?

Sorry, I did not get your question. What do you mean?

Reply:
`spk_man_pair.second->GetOldestKeyPoolTime()` could return `std::nullopt`. Is it well-defined what `std::min` will do here, and is it the desired behaviour?

Reply:
[`std::min()`](https://en.cppreference.com/w/cpp/algorithm/min) uses `operator<`, which is [well-defined](https://en.cppreference.com/w/cpp/utility/optional/operator_cmp) for `std::optional`.